packages/router/src/router.ts

```diff
@@ -372,6 +365,8 @@ export class Router {
       nav = this.navs[name] = new Nav(this, name);
     }
     nav.addRoutes(routes);
+    // tslint:disable-next-line:prefer-object-spread
+    this.navs[name] = Object.assign({}, nav);
```

Comment:
```suggestion
    this.navs[name] = { ...nav };
```

That lint rule is there for a good reason :)

Reply:
This gives me typing errors that I haven't figured out how to resolve. @fkleuver @BBosman

Reply:
Actually I'm amazed this works (and I think it doesn't). An object spread will only copy an object’s own enumerable properties, which explains the typing errors you got, as the missing functions are on the prototype and are thus not copied.

The same applies to `Object.assign` though. And some quick tests I did confirm that in both cases the functions are not copied.

So I think the basic logic has a flaw and Typescript is able to determine that in the case of the object spread, but not in the case of `Object.assign`. Another reason why object spread is preferable. 😉

For reference this is the error you're getting?
`Type '{ name: string; routes: NavRoute[]; router: Router; }' is missing the following properties from type 'Nav': addRoutes, addRoute`

Imho the way to fix this is to do something like this:
```ts
this.navs[name] = new Nav(nav.router, nav.name, nav.routes);
```
And add an (internal?) constructor to also accepts that `router` argument.

Reply:
Thanks for the help! I went with your suggestion and changed the constructor to
```typescript
constructor(router: Router, name: string, routes: NavRoute[] = [])
```